admin-client/pom.xml

```diff
@@ -150,6 +152,34 @@
                                                 org/keycloak/common/util/StringPropertyReplacer.java,
                                                 org/keycloak/common/util/SystemEnvProperties.java,
                                                 org/keycloak/common/util/Time.java,
+                                                org/keycloak/common/crypto/CryptoConstants.java,
+                                                org/keycloak/common/constants/ServiceAccountConstants.java,
+                                            </includes>
+                                        </artifactItem>
+                                        <artifactItem>
+                                            <groupId>org.keycloak</groupId>
+                                            <artifactId>keycloak-server-spi</artifactId>
+                                            <version>${keycloak.version}</version>
+                                            <type>jar</type>
+                                            <classifier>sources</classifier>
+                                            <overWrite>true</overWrite>
+                                            <outputDirectory>${project.build.directory}/unpacked</outputDirectory>
+                                            <includes>
+                                                org/keycloak/utils/StringUtil.java,
+                                            </includes>
+                                        </artifactItem>
+                                        <artifactItem>
+                                            <groupId>org.keycloak</groupId>
+                                            <artifactId>keycloak-server-spi-private</artifactId>
```

Comment:
I am not sure about syncing classes from `keycloak-server-spi-private` as that in theory makes them part of the public API.

For example currently we can remove/update any constants in the class `Constants` in the Keycloak because they are part of the private SPI. But if we move them to the public admin-client library, it makes them exposed to the public and hence we will need to preserve backwards compatibility.

It looks that those classes are needed just in the testsuite. Is it correct? How about just copying them (not syncing) to the testsuite module? If later they change in Keycloak (which is probably not so often), we don't need to necessarily update them right away unless some tests are failing. WDYT?

Reply:
I understand and agree with you, removed the `keycloak-server-spi-private` synchronization and copied the classes into the testsuite